packages/opentelemetry-plugin-user-interaction/test/userInteraction.test.ts

```diff
@@ -0,0 +1,392 @@
+              spanClick.spanContext.spanId,
+              'xhr span has wrong parent'
+            );
+            assertClickSpan(spanClick);
+
+            const attributes = spanXhr.attributes;
+            assert.equal(attributes.component, 'xml-http-request');
+            assert.equal(
+              attributes['http.url'],
+              'https://raw.githubusercontent.com/open-telemetry/opentelemetry-js/master/package.json'
+            );
+            // all other attributes are checked in xhr anyway
+
+            done();
+          });
+        });
+      });
+    });
+
+    it('should ignore interaction when element is disabled', done => {
+      const btn = createButton(true);
+      let called = false;
+      const callback = function() {
+        called = true;
+      };
+      fakeInteraction(callback, btn);
+      sandbox.clock.tick(1000);
+      originalSetTimeout(() => {
+        assert.equal(called, false, 'callback should not be called');
+        done();
+      });
+    });
+
+    it('should handle 3 overlapping interactions', done => {
+      const btn1 = document.createElement('button');
+      btn1.setAttribute('id', 'btn1');
+      const btn2 = document.createElement('button');
+      btn2.setAttribute('id', 'btn2');
+      const btn3 = document.createElement('button');
+      btn3.setAttribute('id', 'btn3');
+      fakeInteraction(() => {
+        getData(FILE_URL, () => {
+          sandbox.clock.tick(10);
+        }).then(() => {});
+      }, btn1);
+      fakeInteraction(() => {
+        getData(FILE_URL, () => {
+          sandbox.clock.tick(10);
+        }).then(() => {});
+      }, btn2);
+      fakeInteraction(() => {
+        getData(FILE_URL, () => {
+          sandbox.clock.tick(10);
+        }).then(() => {});
+      }, btn3);
+      sandbox.clock.tick(1000);
+      originalSetTimeout(() => {
+        assert.equal(exportSpy.args.length, 6, 'should export 6 spans');
+
+        const span1: tracing.ReadableSpan = exportSpy.args[0][0][0];
+        const span2: tracing.ReadableSpan = exportSpy.args[1][0][0];
+        const span3: tracing.ReadableSpan = exportSpy.args[2][0][0];
+        const span4: tracing.ReadableSpan = exportSpy.args[3][0][0];
+        const span5: tracing.ReadableSpan = exportSpy.args[4][0][0];
+        const span6: tracing.ReadableSpan = exportSpy.args[5][0][0];
+
+        assertClickSpan(span1, 'btn1');
+        assertClickSpan(span2, 'btn2');
+        assertClickSpan(span3, 'btn3');
+
+        assert.strictEqual(
+          span1.spanContext.spanId,
+          span4.parentSpanId,
+          'span4 has wrong parent'
+        );
+        assert.strictEqual(
+          span2.spanContext.spanId,
+          span5.parentSpanId,
+          'span5 has wrong parent'
+        );
+        assert.strictEqual(
+          span3.spanContext.spanId,
+          span6.parentSpanId,
+          'span6 has wrong parent'
+        );
+
+        done();
+      });
+    });
+
+    describe('_createSpan', () => {
```

Comment:
Generally I recommend only testing the public API of a class. Could you test this by instead having a dummy span processor that gets added to a tracer with the plugin and observes what spans the plugin creates?

Reply:
Hmm not really because I want to test that specific function easily, without adding more complexity to that especially when using the zone. I would rather make it public instead of adding more complexity for tests if this is something you think it should be done, but personally I don't see any benefit to make it public except for easier testing. WDYT ?

Reply:
Sorry for the slow response here. My thought is still that I would not suggest testing private methods of a class, but rather to figure out how the side-effects of those methods eventually get visible in a public interface, which in this case would be a SpanProcessor for the created span.

I tend to view the public methods as the external contract a class has and so those are the ones we would test, whereas the private methods are the internal implementation details that meet that public contact.

That said, if it feels important to test some functionality here besides via the SpanProcessor, maybe there's an opportunity to extract some kind of utility class or function and test that separately?

Here are a couple references about testing private vs. public methods:
https://enterprisecraftsmanship.com/posts/unit-testing-private-methods/
https://anthonysciamanna.com/2016/02/14/should-private-methods-be-tested.html